src/main/drivers/compass/compass_hmc5883l.c

```diff
@@ -37,13 +38,57 @@
 #include "drivers/sensor.h"
 #include "drivers/time.h"
 
+#include "config/parameter_group.h"
+#include "config/parameter_group_ids.h"
+
 #include "compass.h"
 
 #include "compass_hmc5883l.h"
 #include "compass_spi_hmc5883l.h"
 
 //#define DEBUG_MAG_DATA_READY_INTERRUPT
 
+PG_REGISTER_WITH_RESET_TEMPLATE(busDeviceConfig_t, magHMC5883Config, PG_BUSDEV_HMC5883_CONFIG, 0);
+
+#ifdef USE_MAG_SPI_HMC5883
+#define HMC5883_BUSTYPE BUSTYPE_SPI
+#else
+#define HMC5883_BUSTYPE BUSTYPE_I2C
+#endif
+
+#ifdef MAG_I2C_INSTANCE
+#define HMC5883_BUSNUM MAG_I2C_INSTANCE
+#else
+#define HMC5883_BUSNUM I2CINVALID
+#endif
+
+#ifndef HMC5883_CS_PIN
+#define HMC5883_CS_PIN NONE
+#endif
+
+#if defined(USE_MAG_DATA_READY_SIGNAL) && defined(MAG_INT_EXTI)
+#define HMC5883_INT_EXTI MAG_INT_EXTI
+#else
+#define HMC5883_INT_EXTI NONE
+#endif
+
+#define MAG_ADDRESS 0x1E
+
+#define I2C_DEV_TO_CFG(x) ((x) + 1)
+#define I2C_CFG_TO_DEV(x) ((x) - 1)
+
+PG_RESET_TEMPLATE(busDeviceConfig_t, magHMC5883Config,
+    .busType = HMC5883_BUSTYPE,
+    .busNum = I2C_DEV_TO_CFG(HMC5883_BUSNUM),
+    .i2cAddr = MAG_ADDRESS,
+    .spiCsTag = IO_TAG(HMC5883_CS_PIN),
+    .drdyTag = IO_TAG(HMC5883_INT_EXTI),
+);
+
+#ifndef USE_MAG_SPI_HMC5883
```

Comment:
SPI/I2C should be eventually runtime-configured.

In case of hmc5883l, it should be possible to implement 
```
hmc5883ReadData(Bus_t*, uint8_t reg, void* data, int len);
hmc5883WriteCommand(Bus_t*, uint8_t ref, uint8_t data);
...
```